spec/routing/routes_spec.rb

```diff
@@ -0,0 +1,1590 @@
+        expect(get: grader_path + '/download_grader_criteria_mapping')
+          .to route_to(
+          controller: grader_ctrl,
+          action: 'download_grader_criteria_mapping',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes GET download_dialog properly' do
+        expect(get: grader_path + '/download_dialog').to route_to(
+          controller: grader_ctrl,
+          action: 'download_dialog',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes GET download_grouplist properly' do
+        expect(get: grader_path + '/download_grouplist').to route_to(
+          controller: grader_ctrl,
+          action: 'download_grouplist',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes GET grader_criteria_dialog properly' do
+        expect(get: grader_path + '/grader_criteria_dialog').to route_to(
+          controller: grader_ctrl,
+          action: 'grader_criteria_dialog',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+      
+      it 'routes GET global_actions properly' do
+        expect(get: grader_path + '/global_actions').to route_to(
+          controller: grader_ctrl,
+          action: 'global_actions',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes GET groups_coverage_dialog properly' do
+        expect(get: grader_path + '/groups_coverage_dialog').to route_to(
+          controller: grader_ctrl,
+          action: 'groups_coverage_dialog',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes POST set_assign_criteria properly' do
+        expect(post: grader_path + '/set_assign_criteria').to route_to(
+          controller: grader_ctrl,
+          action: 'set_assign_criteria',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+      
+      it 'routes GET random_assign properly' do
+        expect(get: grader_path + '/random_assign').to route_to(
+          controller: grader_ctrl,
+          action: 'random_assign',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes GET upload_dialog properly' do
+        expect(get: grader_path + '/upload_dialog').to route_to(
+          controller: grader_ctrl,
+          action: 'upload_dialog',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+
+      it 'routes POST global_actions properly' do
+        expect(post: grader_path + '/global_actions').to route_to(
+          controller: grader_ctrl,
+          action: 'global_actions',
+          assignment_id: assignment.id.to_s,
+          locale: 'en')
+      end
+    end
+  end
+  # end assignment graders route tests
+
+  context 'annotation_categories' do
+    let(:annot_path) {
```

Comment:
I have researched and there seems to be no way to re-write 'let' with 'do end' syntax